packages/react/src/components/Authenticator/SignUp/FormFields.tsx

```diff
@@ -1,82 +1,217 @@
 import {
   getActorContext,
   LoginMechanism,
-  LoginMechanismArray,
   SignUpContext,
   translate,
 } from '@aws-amplify/ui';
-import { isEmpty } from 'lodash';
 
 import { useAuthenticator } from '..';
-import { PasswordField, Text } from '../../..';
+import { PasswordField, PhoneNumberField, Text, TextField } from '../../..';
 import { UserNameAlias as UserNameAliasComponent } from '../shared';
 
 export function FormFields() {
   const { _state } = useAuthenticator();
-  const { validationError } = getActorContext(_state) as SignUpContext;
-
-  const [primaryAlias, ...secondaryAliases] =
-    _state.context.config?.loginMechanisms?.filter(
-      (alias: any): alias is LoginMechanism =>
-        LoginMechanismArray.includes(alias)
-    ) ?? LoginMechanismArray;
-
-  /**
-   * If the login_mechanisms are configured to use ONLY username, we need
-   * to ask for some sort of secondary contact information in order to
-   * verify the user for Cognito. Currently matching this to how Vue is
-   * set up.
-   */
-  if (primaryAlias === 'username' && isEmpty(secondaryAliases)) {
-    secondaryAliases.push('email', 'phone_number');
-  }
-
-  const passwordLabel = translate('Password');
-  const confirmPasswordLabel = translate('Confirm Password');
-  const passwordFieldClass = 'password-field';
+  const { country_code, validationError } = getActorContext(
+    _state
+  ) as SignUpContext;
+  const { loginMechanisms, signUpAttributes } = _state.context.config;
+
+  const fieldNames = Array.from(
+    new Set([...loginMechanisms, ...signUpAttributes])
+  );
+
+  // Only 1 is supported, so `['email', 'phone_number']` will only show `email`
+  const loginMechanism = fieldNames.shift() as LoginMechanism;
 
   return (
     <>
-      <UserNameAliasComponent data-amplify-usernamealias alias={primaryAlias} />
+      <UserNameAliasComponent
+        alias={loginMechanism}
+        data-amplify-usernamealias
+      />
 
       <PasswordField
+        autoComplete="new-password"
         data-amplify-password
-        className={passwordFieldClass}
-        placeholder={passwordLabel}
-        isRequired={true}
+        hasError={!!validationError['confirm_password']}
+        isRequired
         name="password"
-        label={passwordLabel}
+        label={translate('Password')}
         labelHidden={true}
-        autoComplete="new-password"
-        hasError={!!validationError['confirm_password']}
+        placeholder={translate('Password')}
       />
+
       <PasswordField
-        data-amplify-confirmpassword
-        className={passwordFieldClass}
-        placeholder={confirmPasswordLabel}
-        isRequired={true}
-        name="confirm_password"
-        label={confirmPasswordLabel}
-        labelHidden={true}
         autoComplete="new-password"
+        data-amplify-confirmpassword
```

Comment:
Small nit, but we can start cleaning up these unused `data-` attributes after GA.

Reply:
@wlee221 I'd recommend adding a track to task this – this could become a risk for customers to depend on for CSS 😬